sqlmesh/core/scheduler.py

```diff
@@ -380,26 +374,110 @@ def run(
 
         return not errors
 
-    def run_batches(
+    def batch_intervals(
+        self,
+        merged_intervals: SnapshotToIntervals,
+        start: t.Optional[TimeLike] = None,
+        end: t.Optional[TimeLike] = None,
+        execution_time: t.Optional[TimeLike] = None,
+    ) -> t.Dict[Snapshot, Intervals]:
+        def expand_range_as_interval(
+            start_ts: int, end_ts: int, interval_unit: IntervalUnit
+        ) -> t.List[Interval]:
+            values = expand_range(start_ts, end_ts, interval_unit)
+            return [(values[i], values[i + 1]) for i in range(len(values) - 1)]
+
+        dag = DAG[str]()
+
+        for snapshot in merged_intervals:
+            dag.add(snapshot.name, [p.name for p in snapshot.parents])
+
+        snapshot_intervals = {
+            snapshot: list(
+                flatten(
```

Comment:
Do we really need this dependency? Doesn't python do this natively?
```
i
for interval in intervals
for in expand_range_as_interval(*interval, snapshot.node.interval_unit)
```

Reply:
yea i agree, we should get rid of it

Reply:
removed